src/cpu/x64/cpu_isa_traits.hpp

```diff
@@ -477,6 +477,11 @@ inline bool isa_has_bf16(cpu_isa_t isa) {
     return is_superset(isa, avx512_core_bf16);
 }
 
+inline bool isa_has_f16(cpu_isa_t isa) {
+    return is_superset(isa, avx512_core_fp16)
+            || is_superset(isa, avx10_1_512_amx_fp16);
```

Comment:
I think `avx2_vnni_2` has f16, at least available instructions for conversion at least. For scales that's more than enough to be correctly processed.